src/book/03-guides/04-general/02-troubleshooting.mdx

```diff
@@ -35,6 +35,31 @@ Run `./b configure --clean`. This refreshes the docker volumes, removing the off
 
 If you have just added a new file to the codebase, make sure that it's part of a role. After you have added it to a role, run `./b configure`, and the build system should detect the new file and compile it the next time you run `./b build`.
 
+### Why am I having CMakeCache errors when I do `./b run <role>`?
+
+If you're having errors which say that there was no such value in `b.cmake_cache` such as in this snippet try running `./b configure --clean` then `./b build`.
+
+```py
+Traceback (most recent call last):
+  File "/home/nubots/NUbots/nuclear/b.py", line 113, in <module>
+    module.run(**vars(command.parse_args()))
+  File "/home/nubots/NUbots/tools/utility/dockerise/run.py", line 79, in _run
+    func(**kwargs)
+  File "/home/nubots/NUbots/tools/run.py", line 34, in run
+    use_asan = b.cmake_cache["USE_ASAN"] == "ON"
+KeyError: 'USE_ASAN'
+```
+
+If the problem persists, you probably have an extra `CMakeCache.txt` file in your NUbots directory, which is messing up the build's cache.
+Fixing this issue is as simple as finding the offending file and deleting it.
+You can find the file in your text editor, or by running the find command inside the NUbots directory:
```

Comment:
I would do this first. The existence of this file (and the `CMakeFiles` folder) is indicative of a larger problem and removing these will likely resolve other issues that are being experienced. If removing these doesnt fix anything, then I would purge the build volume (`./b configure --clean`)